File: packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/cozy/CozyMarkerElements.java
package io.flutter.plugins.googlemaps.cozy;

public class CozyMarkerElements {
    public final CozyMarkerElement canvas;
    public final CozyMarkerElement bubble;
    public final CozyMarkerElement[] labels;
    public final CozyMarkerElement[] icons;
    public final CozyMarkerElement iconCircle;
    public final CozyMarkerElement pointer;
    public final CozyMarkerElement counterBubble;

    public CozyMarkerElements(CozyMarkerElement canvas, CozyMarkerElement bubble, CozyMarkerElement counterBubble, CozyMarkerElement[] labels, CozyMarkerElement[] icons, CozyMarkerElement iconCircle, CozyMarkerElement pointer) {
        this.canvas = canvas;
        this.bubble = bubble;
        this.counterBubble = counterBubble;
        this.labels = labels;
        this.icons = icons;
        this.iconCircle = iconCircle;
        this.pointer = pointer;
    }

    @Override
    public String toString() {
        return canvas.toString() + bubble.toString() + counterBubble.toString() + labels.toString() + icons.toString() + iconCircle.toString() + pointer.toString();
    }
}

File: packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/CozyMarkerElements.m
//
//  CozyMarkerElements.m
//  Pods
//
//  Created by Pietro Domingues on 26/09/23.
//

#import "CozyMarkerElements.h"

@implementation CozyMarkerElement
- (instancetype)initWithBounds:(CGRect)bounds fillColor:(UIColor *)fillColor strokeColor:(UIColor *)strokeColor alpha:(CGFloat)alpha data:(NSObject *)data {
    self = [super init];
    if (self) {
        _bounds = bounds;
        _fillColor = fillColor;
        _strokeColor = strokeColor;
        _alpha = alpha;
        _data = data;
    }
    return self;
}
@end

@implementation CozyMarkerElements
- (instancetype)initWithCanvas:(CozyMarkerElement *)canvas bubble:(CozyMarkerElement *)bubble labels:(NSArray<CozyMarkerElement *> *)labels counter:(CozyMarkerElement *)counter icon:(CozyMarkerElement *)icon iconCircle:(CozyMarkerElement *)iconCircle pointer:(CozyMarkerElement *)pointer {
    self = [super init];
    if (self) {
        _canvas = canvas;
        _bubble = bubble;
        _counter = counter;
        _icon = icon;
        _iconCircle = iconCircle;
        _pointer = pointer;
        _labels = labels;
    }
    return self;
}
@end
